airflow/contrib/operators/gcp_compute_operator.py

```diff
@@ -0,0 +1,354 @@
+        if validate_body:
+            self._field_validator = BodyFieldValidator(
+                SET_MACHINE_TYPE_VALIDATION_SPECIFICATION, api_version=api_version)
+        self._validate_inputs()
+        self._hook = GceHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        super(GceSetMachineTypeOperator, self).__init__(*args, **kwargs)
+
+    def _validate_all_body_fields(self):
+        if self._field_validator:
+            self._field_validator.validate(self.body)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is missing")
+        if not self.zone:
+            raise AirflowException("The required parameter 'zone' is missing")
+        if not self.resource_id:
+            raise AirflowException("The required parameter 'resource_id' is missing")
+
+    def execute(self, context):
+        self._validate_all_body_fields()
+        return self._hook.set_machine_type(self.project_id, self.zone,
+                                           self.resource_id, self.body)
+
+
+class GceInstanceGroupGetOperator(BaseOperator):
+    """Gets information about instance group.
+
+    TODO: What to do with the information retrieved ?
+
+    :param project_id: Project ID that the operator works on
+    :type project_id: str
+    :param zone: Zone where the operator operates on
+    :type zone: str
+    :param resource_id: Name of the instance resource
+    :type resource_id: str
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: Version of the API used (for example v1).
+    :type api_version: str
+    """
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 zone,
+                 resource_id,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v1',
+                 *args, **kwargs):
+        self.project_id = project_id
+        self.zone = zone
+        self.full_location = 'projects/{}/zones/{}'.format(self.project_id,
+                                                           self.zone)
+        self.resource_id = resource_id
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self._hook = GceHook(gcp_conn_id=self.gcp_conn_id, api_version=self.api_version)
+        self._validate_inputs()
+        super(GceInstanceGroupGetOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is missing")
+        if not self.zone:
+            raise AirflowException("The required parameter 'zone' is missing")
+
+    def execute(self, context):
+        # TODO
+        pass
+
+
+INSTANCE_GROUP_VALIDATION_SPECIFICATION = [
+    # TODO: Add the validation spec
+]
+
+
+class GceInstanceGroupDeployOperator(BaseOperator):
```

Comment:
We should separate those three group operators to a separate commit (now that we know we will make separate PRs